p4_symbolic/symbolic/symbolic.cc

```diff
@@ -0,0 +1,126 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "p4_symbolic/symbolic/symbolic.h"
+
+#include "p4_symbolic/symbolic/table.h"
+#include "p4_symbolic/symbolic/util.h"
+
+namespace p4_symbolic {
+namespace symbolic {
+
+z3::context *Z3_CONTEXT = nullptr;
+
+pdpi::StatusOr<SolverState *> EvaluateP4Pipeline(
```

Comment:
Is there are reason to return the pointer? Why not just return the value?

Reply:
Yes, the SolverState has a z3::context and z3::solver in it. These do not have copy constructors. pdpi::StatusOr<> requires a copy constructor. Also, because I wanted SolverState to be essentially a singleton.